messaging-client-databus/src/main/java/com/inmobi/databus/partition/CollectorReader.java

```diff
@@ -262,4 +263,32 @@ public MessageCheckpoint getMessageCheckpoint() {
   public MessageCheckpoint buildStartPartitionCheckpoints() {
     return null;
   }
+
+  @Override
+  public synchronized Long getReaderBackLog() throws Exception {
+    Long pendingSize = 0l;
+    int timedWaiting = 0;
+    int waitingThreshold = 10;
+    while (cReader.getCurrentFile() == null && timedWaiting < waitingThreshold){
+      Thread.sleep(1000);
+      timedWaiting++;
+    }
+    //get collector reader remaining size
+    if (cReader.getCurrentFile() != null){
+      pendingSize += cReader.getPendingSize(cReader.getCurrentFile());
+      LOG.info("Pending Size inside collector reader - collector added "+pendingSize);
+    }
+    //get local reader remaining size
+    timedWaiting = 0;
+    while (lReader.getCurrentFile() == null && timedWaiting < waitingThreshold){
+      Thread.sleep(1000);
+      timedWaiting++;
+    }
+    if (lReader.getCurrentFile() != null){
+      pendingSize += lReader.getPendingSize();
+      LOG.info("Pending Size inside collector reader - local added "+pendingSize);
```

Comment:
LOG.info("Pending Size inside streams_local output location starting from file: " +  lReader.getCurrentFile() + " is  + pendingSize);